### PR TITLE
0.3.1 with bug fixes

### PR DESCRIPTION
**Pull request by hawkfalcon**

### CODE REVIEW

File: TimberAPI/AssetLoaderSystem/AssetSystem/AssetRepository.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public CustomAssetBundle FindByPathAndFileName(string[] path, string fileName)
         {
             CustomAssetBundle bundle = _assetBundles.FirstOrDefault(bundle => bundle.AssetPath.SequenceEqual(path) && bundle.FileName == fileName);
             if (bundle == null)
-                throw new AssetNotFoundException();
+                throw new AssetNotFoundException(fileName);
```
Comment (hytonhan) [inline TimberAPI/AssetLoaderSystem/AssetSystem/AssetRepository.cs:25]: Just a sidenote, imo a cool way to throw errors in cases like these:
```c#
var bundle = _assetBundles.FirstOrDefault(bundle => bundle.AssetPath.SequenceEqual(path) && bundle.FileName == fileName) ?? throw new AssetNotFoundException(fileName);
return bundle;
```

**Review by hytonhan — APPROVED** [review]
Looks good to me, left some opinions

File: TimberAPI/AssetLoaderSystem/AssetRegistry.cs [hunk 2]
```diff
@@ -19,18 +19,22 @@ public class AssetRegistry : IAssetRegistry
...
-            AddSceneAssets(prefix, assetEntryPoint, new[] { "assets" });
+            AddSceneAssets(prefix, assetEntryPoint, new[] { "assets" }, Assembly.GetCallingAssembly()?.Location);
         }
 
         public void AddSceneAssets(string prefix, SceneEntryPoint assetEntryPoint, string[] assetLocation)
         {
+            AddSceneAssets(prefix, assetEntryPoint, assetLocation, Assembly.GetCallingAssembly()?.Location);
+        }
+
+        public void AddSceneAssets(string prefix, SceneEntryPoint assetEntryPoint, string[] assetLocation, string assemblyLocation) {
```
[the remark this answers is not in the document]
Comment (KYPremco) [reply TimberAPI/AssetLoaderSystem/AssetRegistry.cs:35]: Sure :)

File: TimberAPI/AssetLoaderSystem/AssetSystem/AssetRepository.cs (hunk 1 above)
Comment (KYPremco) [reply TimberAPI/AssetLoaderSystem/AssetSystem/AssetRepository.cs:25]: Prefer the chose way faster readable.
Im not sure but I thought there was something so you can skip the if statement? and throw when its null directly.
Not an needed change tho

**Review by KYPremco — APPROVED** [review]
You can mergeit if you don't want to change that one part (if even possible)

File: TimberAPI/AssetLoaderSystem/AssetRegistry.cs (hunk 2 above)
Comment (hytonhan) [inline TimberAPI/AssetLoaderSystem/AssetRegistry.cs:35]: Should this overload also be added to the IAssetRegistry interface?

**Pull request merged**